server/src/main/java/org/elasticsearch/action/fieldcaps/IndexFieldCapabilities.java

```diff
@@ -62,7 +60,7 @@ public class IndexFieldCapabilities implements Writeable {
     IndexFieldCapabilities(StreamInput in) throws IOException {
         if (in.getVersion().onOrAfter(Version.V_7_7_0)) {
             this.name = in.readString();
-            this.type = typeStringDeduplicator.deduplicate(in.readString());
+            this.type = in.readString();
```

Comment:
I get where this is coming from and like the idea here a lot, but isn't type still going to be a massively repeated string potentially if we have a ton of different fields in a large use-case? Might still be worthwhile to dedup this?

Reply:
I think we should keep the type deduplicator. I restored it in https://github.com/elastic/elasticsearch/pull/88238/commits/4b9c1aad17746ddfd21192b0d66ce89e3450a70f.